tests/unit_tests/scheduler/test_scheduler.py

```diff
@@ -204,6 +205,44 @@ async def wait():
     assert timeouteventfound
 
 
+@pytest.mark.parametrize("max_running", [0, 1, 2, 10])
+async def test_max_running(max_running, mock_driver, storage, tmp_path):
+    runs: List[bool] = []
+
+    async def wait():
+        nonlocal runs
+        runs.append(True)
+        await asyncio.sleep(0.01)
+        runs.append(False)
+
+    ensemble_size = max_running * 3 if max_running > 0 else 10
+
+    ensemble = storage.create_experiment().create_ensemble(
+        name="foo", ensemble_size=ensemble_size
+    )
+    realizations = [
+        create_stub_realization(ensemble, tmp_path, iens)
+        for iens in range(ensemble_size)
+    ]
+
+    sch = scheduler.Scheduler(
+        mock_driver(wait=wait), realizations, max_running=max_running
+    )
+
+    await sch.execute()
```

Comment:
Just to be sure: ` assert await sch.execute() == EVTYPE_ENSEMBLE_STOPPED` ?

Reply:
Can do, but it adds assertions that are not related to the purpose of the test. Can it cause unneccesary maintenance later on to have bonus asserts?